CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -287,6 +311,57 @@ local function CreateMenuBar(barAlignment)
 	return this
 end
 
+local function Create3DMenuBar(barAlignment, threeDPanel)
+	local this = CreateMenuBar(barAlignment)
+
+	local superArrangeItems = this.ArrangeItems
+	function this:ArrangeItems()
+		local totalWidth = superArrangeItems(self)
+		if threeDPanel then
+			threeDPanel:ResizePixels(totalWidth, TOPBAR_THICKNESS)
+		end
+		return totalWidth
+	end
+
+	if threeDPanel then
+		local RENDER_STEP_NAME = game:GetService("HttpService"):GenerateGUID() .. "Create3DMenuBar"
+		threeDPanel:AddTransparencyCallback(function(transparency)
+			for _, item in pairs(this:GetItems()) do
+				item:SetTransparency(transparency)
+			end
+		end)
+
+		local lastHoveredItem = nil
+		local function OnRenderStep()
+			local hoveredItem = Panel3D.FindHoveredGuiElement(threeDPanel, this:GetItems())
```

Comment:
`	function this:GetItems()
 		return items
 	end`

Just returns the table, we could save it in a local variable so we don't have to call the function again.